app/services/match_creator.rb

```diff
@@ -6,30 +8,11 @@ def initialize(tournament, registered_users, round = 0)
     @round = round
   end
 
-  def create_match
-    data = group_by_two
-    matches = data[1]
-    user_matches = data[0]
-    Match.transaction do
-      user_matches.each(&:save!)
-    end
-    matches.each(&:delayed_job)
-  end
-
-  def group_by_two
-    user_matches = []
-    matches = []
-    match_time = @tournament.current_match_time
+  def call
+    time = @tournament.current_match_time
```

Comment:
This is just an idea:
```suggestion
     raise StandardError("Users can't be less than 8") if @tournament.users 
     
    time = @tournament.current_match_time
```